scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+
+      val repoPath = args[0] + "/"
+
+      val allowedDirectories: MutableList<String> = ArrayList()
+
+      for (layerIndex in 1 until args.size) {
+        allowedDirectories.add(args[layerIndex])
+      }
+
+      val searchFiles = collectSearchFiles(repoPath, allowedDirectories)
+
+      var scriptFailedFlag = false
+
+      getFilenameChecks().forEach {
+        if (checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameRegexString = it.getProhibitedFilenameRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      getFileContentChecks().forEach {
+        if (checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            fileNameRegexString = it.getFilenameRegex(),
+            prohibitedContentRegexString = it.getProhibitedContentRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      if (scriptFailedFlag) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+    }
+
+    /**
+     * Fetches all filename checks
+     *
+     * @return [List<FilenameCheck>] a list of all the FilenameChecks
+     */
+    fun getFilenameChecks(): List<FilenameCheck> {
+      val fileNamePatternsBinaryFile =
+        File("scripts/assets/filename_pattern_validation_checks.pb")
+      val filenameCheckBuilder = FilenameChecks.newBuilder()
+      val namePatternsObj: FilenameChecks =
+        FileInputStream(fileNamePatternsBinaryFile).use {
+          filenameCheckBuilder.mergeFrom(it)
+        }.build() as FilenameChecks
+
+      return namePatternsObj.getFilenameChecksList()
+    }
+
+    /**
+     * Fetches all file content checks
+     *
+     * @return [List<FileContentCheck>] a list of all the FileContentChecks
+     */
+    fun getFileContentChecks(): List<FileContentCheck> {
+      val fileContentsBinaryFile =
+        File("scripts/assets/file_content_validation_checks.pb")
+      val fileContentCheckBuilder = FileContentChecks.newBuilder()
+      val fileContentsObj: FileContentChecks =
+        FileInputStream(fileContentsBinaryFile).use {
+          fileContentCheckBuilder.mergeFrom(it)
+        }.build() as FileContentChecks
+
+      return fileContentsObj.getFileContentChecksList()
+    }
+
+    /**
+     * Checks for a prohibited file naming pattern
+     *
+     * @param repoPath the path of the repo.
+     * @param searchFiles a list of all the files which needs to be checked.
+     * @param fileNameRegexString filename pattern regex which should not be present
+     * @param errorToShow error to show incase of failure
+     * @return [Boolean] check failed or passed
```

Comment:
The return type isn't needed here (& usually not included): ``@return whether the check has passed`` is sufficient.

Reply:
removed, and will remember not to include it next time.